sdk/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -220,24 +220,42 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
+const (
+	ResourceSelectorAll = "*"
+)
+
+// ResourceSelectorName validates a string for use with a PermissionClaim.
+//
+// +kubebuilder:validation:Type=string
+// +kubebuilder:validation:MinLength=1
+type ResourceSelectorName string
+
+// ResourceSelector identifies the objects to be included within a PermissionClaim either by names and/or namespaces.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.namespaces) || has(self.names)",message="at least one field must be set"
 type ResourceSelector struct {
-	// name of an object within a claimed group/resource.
-	// It matches the metadata.name field of the underlying object.
-	// If namespace is unset, all objects matching that name will be claimed.
+	// names is a list of specific resources to select.
+	//
+	// Names matches the metadata.name field of the underlying object.
+	// An entry of "*" anywhere in the list means all object names of the group/resource within the "namespaces" field are claimed.
+	// Wildcard entries other than "*" and regular expressions are currently unsupported.
+	// Default is "*".
 	//
 	// +optional
-	// +kubebuilder:validation:Pattern="^([a-z0-9][-a-z0-9_.]*)?[a-z0-9]$"
-	// +kubebuilder:validation:MaxLength=253
-	// +kubebuilder:validation:MinLength=1
-	Name string `json:"name,omitempty"`
+	// +kubebuilder:default={"*"}
+	Names []ResourceSelectorName `json:"names,omitempty"`
 
-	// namespace containing the named object. Matches metadata.namespace field.
-	// If "name" is unset, all objects from the namespace are being claimed.
+	// namespaces represents namespaces where an object of the given group/resource may be managed.
+	//
+	// Namespaces matches against metadata.namespace field.
+	// A value of "*" matches namespaced objects across all namespaces.
+	// If Namespaces is not set (an empty list), matches cluster-scoped resources.
```

Comment:
Let's describe the exact states "If Namespaces is not set or set to empty list, matches cluster-scoped resources.